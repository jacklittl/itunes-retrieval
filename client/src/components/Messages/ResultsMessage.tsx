import React from "react";
import { useSelector } from "react-redux";

import Typography from '@material-ui/core/Typography';

const ResultsMessage = (): JSX.Element => {
  const results = useSelector(state => state.form)

  let searchTitle:string;

  results.error 
    ? searchTitle = "Error retrieving your results."
    : results.loading 
      ? searchTitle = "Finding your results..."
      : results.data.payload && results.data.payload.length === 0 
        ? searchTitle = "No results found..."
        : searchTitle = "iTunes Search:"

  return (
    <div>
      <Typography gutterBottom variant="h5" color="primary" component="h2">{searchTitle}</Typography>
      <Typography variant="body2" color="textSecondary" component="p">Please search for your favourite bands, artists, movies etc above. Alternatively, if no results could be found, please try altering you search.</Typography>
    </div>
  )
};

export default ( ResultsMessage )