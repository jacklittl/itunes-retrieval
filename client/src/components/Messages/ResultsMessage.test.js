import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

import ResultsMessage from './ResultsMessage';

const mockStore = configureStore([]);
const store = mockStore({
  form: {
    data: {},
    error: false,
    loading: false,
    success: false
  }
});
const loadingStore = mockStore({
  form: {
    data: {},
    error: false,
    loading: true,
    success: false
  }
});
const errorStore = mockStore({
  form: {
    data: {},
    error: true,
    loading: false,
    success: false
  }
});
const successStore = mockStore({
  form: {
    data: {},
    error: false,
    loading: false,
    success: true
  }
});

configure({adapter: new Adapter()});

describe('Results Message Render', () => {
  test('Results Message Renders', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ResultsMessage />
      </Provider>
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Results Message Text Render', () => {
  it('Has Correct Search Text On Page Load', () => {
    const { getByText } = render(
      <Provider store={store}>
        <ResultsMessage />
      </Provider>
    );
    const title = getByText('iTunes Search:');
    expect(title).toBeInTheDocument();
  });
  it('Has Correct Search Text On Loading', () => {
    const { getByText } = render(
      <Provider store={loadingStore}>
        <ResultsMessage />
      </Provider>
    );
    const title = getByText('Finding your results...');
    expect(title).toBeInTheDocument();
  });
  it('Has Correct Search Text On Error', () => {

    const { getByText } = render(
      <Provider store={errorStore}>
        <ResultsMessage />
      </Provider>
    );
    
    const title = getByText('Error retrieving your results.');
    expect(title).toBeInTheDocument();
  });
  it('Has Correct Search Text On Success', () => {

    const { getByText } = render(
      <Provider store={successStore}>
        <ResultsMessage />
      </Provider>
    );
    
    const title = getByText('iTunes Search:');
    expect(title).toBeInTheDocument();
  });
});