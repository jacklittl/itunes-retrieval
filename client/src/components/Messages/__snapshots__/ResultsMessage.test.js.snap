// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Message Render Results Message Renders 1`] = `
<div>
  <h2
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    iTunes Search:
  </h2>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
  >
    Please search for your favourite bands, artists, movies etc above. Alternatively, if no results could be found, please try altering you search.
  </p>
</div>
`;

exports[`Results Message Render snapshot renders 1`] = `
<div>
  <h2
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    iTunes Search:
  </h2>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
  >
    Please search for your favourite bands, artists, movies etc above. Alternatively, if no results could be found, please try altering you search.
  </p>
</div>
`;
